mypy/constraints.py

```diff
@@ -349,6 +383,11 @@ def visit_instance(self, template: Instance) -> List[Constraint]:
                 cb = infer_constraints(template.args[0], item, SUPERTYPE_OF)
                 res.extend(cb)
             return res
+        elif isinstance(actual, TupleType) and template.type.is_protocol:
```

Comment:
Support also structural checks for ``TypedDict``s via fallback (here and in ``subtypes,py``).